system/Database/Forge.php

```diff
@@ -302,14 +302,12 @@ public function dropDatabase(string $dbName): bool
      *
      * @return Forge
      */
-    public function addKey($key, bool $primary = false, bool $unique = false)
+    public function addKey($key, bool $primary = false, bool $unique = false, string $keyName = '')
     {
         if ($primary) {
-            foreach ((array) $key as $one) {
-                $this->primaryKeys[] = $one;
-            }
+            $this->primaryKeys = ['fields' => (array) $key, 'keyName' => $this->db->protectIdentifiers($keyName)];
```

Comment:
You've changed the `$primaryKeys` structure. Can you add `@phpstan-var` detailed array type in the property?
Something like this?
```
array{fields: string[], keyName: string}
```